lib/events.js

```diff
@@ -634,6 +635,30 @@ function unwrapListeners(arr) {
   return ret;
 }
 
+function getEventListeners(emitterOrTarget, type) {
+  // First check if EventEmitter
+  if (typeof emitterOrTarget.listeners === 'function') {
+    return emitterOrTarget.listeners(type);
+  }
+  // Require event target lazily to avoid always loading it
+  const { isEventTarget, kEvents } = require('internal/event_target');
+  if (isEventTarget(emitterOrTarget)) {
+    const root = emitterOrTarget[kEvents].get(type);
+    const listeners = [];
+    if (root === undefined || root.next === undefined)
+      return [];
+    let handler = root.next;
+    while (handler !== undefined && handler.listener !== undefined) {
```

Comment:
This could be simplified a bit more BTW
```suggestion
    let handler = root?.next;
    while (handler?.listener !== undefined) {
```

Reply:
I think the neatest thing to do (in general) is to make an iterator and then use it in EventTarget internally as well as here.

In the meantime I'll apply your suggestions.